sqlx-core/src/connection.rs

```diff
@@ -172,4 +174,8 @@ pub trait ConnectOptions: 'static + Send + Sync + FromStr<Err = Error> + Debug {
         self.log_statements(LevelFilter::Off)
             .log_slow_statements(LevelFilter::Off, Duration::default())
     }
+
+    /// Returns the driver used for the connection.
+    #[cfg(feature = "any")]
+    fn get_kind(&self) -> AnyKind;
```

Comment:
This is technically a backwards-incompatible change and so will need to be part of 0.6.0

Reply:
Additive changes are backward-compatible, unless you're describing trait-ABI compatibility? Since there is no official ABI for Rust yet, I don't think we need to consider that?

Reply:
You could just have an inherent method `kind(&self) -> AnyKind` on `AnyConnectOptions` and then an inherent method on `Pool<Any>` that forwards to that. Would that suit your use case?

Reply:
Might be worth just releasing it as a breaking-change; it's not like it will benefit the performance semantics for anyone who would receive a patch update.

Reply:
So what's the way forward? Should we release this as a breaking change, or implement @abonander's proposal? (sorry I lost a bit of context on that PR over the months so I'd have to get back in to understand his proposal)

Reply:
I would prefer to just have it be an inherent method on `Pool<Any>` if that suits the use-case. I don't think it's necessary to change a fundamental trait like this.